python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -903,24 +903,13 @@ def sample_negative_edges_uniform(
             `edge_type`. Note that negative refers to false negatives, which
             means the edge could be present or not present in the graph.
         """
-        if edge_type is not None:
-            assert (
-                self.node_type_offset is not None
-            ), "The 'node_type_offset' array is necessary for performing \
-                negative sampling by edge type."
-            _, _, dst_node_type = etype_str_to_tuple(edge_type)
-            dst_node_type_id = self.node_type_to_id[dst_node_type]
-            offset = self._node_type_offset_list
-            max_node_id = (
-                offset[dst_node_type_id + 1] - offset[dst_node_type_id]
-            )
+        if edge_type:
```

Comment:
as no c++ dependency on `FusedCSCSamplingGraph` now, why not implement the negative sampling in `UniformNegativeSampler::_sample_with_etype()` directly?

Reply:
It need information from graph.

Reply:
graph is already passed into negative sampler